### PR TITLE
Remove access to wlr_input_device union

### PR DESCRIPTION
**Pull request by emersion**
References: https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/3626
Closes: https://github.com/swaywm/sway/issues/7077

### CODE REVIEW

**Comment by J0nnyMak0** [comment]
Works for me. Thanks!

**Comment by MisterSincere** [comment]
What persons should give approving reviews? I was just doing the same changes and the changes look good and quiet simple to me, but I never looked into sway's code before, which is why I didn't want to give approval.

**Review by bl4ckb0ne — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
